crates/slot-arithmetic/src/lib.rs

```diff
@@ -12,10 +12,31 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use derive_more::{Add, From, Sub};
 use minicbor::{Decode, Decoder, Encode};
 use serde::{Deserialize, Serialize};
 
-type Slot = u64;
+#[derive(Clone, Debug, Copy, PartialEq, From, Add, Sub, PartialOrd, Eq, Serialize, Deserialize)]
+#[repr(transparent)]
+pub struct Slot {
+    pub slot: u64,
+}
```

Comment:
```suggestion
pub struct Slot(64);
```
The wrapped value should not be directly accessible from the outside. This is more important when guaranteeing additional constraints, in which case no external mutation can be allowed (but a `.get()` or `.as_u64()` still makes sense).

Reply:
I guess you meant tuple-struct, right?